src/main/java/au/com/grieve/multibridge/util/SimpleTemplate.java

```diff
@@ -52,14 +52,24 @@ public String replace(String input) {
      * Given an inFile and outFile, replace all placeholders in inFile and save to Outfile
      */
     public void replace(Path inFile, Path outFile) throws IOException {
-        BufferedWriter writer = new BufferedWriter(new FileWriter(outFile.toFile()));
-        BufferedReader reader = new BufferedReader(new FileReader(inFile.toFile()));
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(outFile.toFile()));
+            BufferedReader reader = new BufferedReader(new FileReader(inFile.toFile()));
 
-        for (String line; ((line = reader.readLine()) != null); ) {
-            writer.write(replace(line));
-            writer.newLine();
+            for (String line; ((line = reader.readLine()) != null); ) {
+                writer.write(replace(line));
+                writer.newLine();
+            }
+        } catch (MissingVariable e) {
+            System.err.println(inFile + ": " + e.getMessage());
```

Comment:
Instead of System.err/System.out best practice now is the plugins logger (Plugin::getLogger()) so it may be better for the caller to catch the exception and it then logging it as it has access to Plugin.

Reply:
I agree that using a logger is better, but I matched the existing code in using `System.err`.

Happy to fix up all occurrences to use a logger, but that might be better suited for a dedicated PR along side other modernisation/best-practice changes (I'm happy to do that too).

Reply:
If it matches existing then that sounds fine and as you say changing it under a separate task would make more sense.